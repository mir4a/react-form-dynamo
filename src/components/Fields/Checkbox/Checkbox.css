.Checkbox {
  margin-bottom: 25px;
}