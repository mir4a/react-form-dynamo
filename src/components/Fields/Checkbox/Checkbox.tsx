import React, { memo, useCallback, ChangeEvent, useState } from 'react';
import classNames from 'classnames';
import Label from '../Label';
import './Checkbox.css';

interface Props {
  initialValue?: string;
  label: string;
  name?: string;
  onChange?: Function;
  className?: string;
}

export default memo(function Checkbox(props: Props) {
  const { initialValue, label, onChange, name, className } = props;
  const [checked, setChecked] = useState(!!initialValue);
  const checkboxClassNames = classNames('Checkbox', className);
  const handleOnChange = useCallback(
    (event: ChangeEvent<HTMLInputElement>) => {
      setChecked(event.target.checked);
      onChange && onChange(event.target.checked);
    },
    [setChecked, onChange],
  );

  return (
    <>
      <Label label={label}>
        <input
          type="checkbox"
          name={name}
          onChange={handleOnChange}
          checked={checked}
          className={checkboxClassNames}
        />
      </Label>
    </>
  );
});
