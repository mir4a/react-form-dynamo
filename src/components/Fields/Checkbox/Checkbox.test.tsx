import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Checkbox initialValue="1" label="Radio Buttons" onChange={() => {}} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
