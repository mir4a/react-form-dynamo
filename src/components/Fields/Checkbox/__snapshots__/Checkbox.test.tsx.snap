// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="Label"
>
  <input
    checked={true}
    className="Checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="Label-text"
  >
    Radio Buttons
  </span>
</label>
`;
